### PR TITLE
[Realtek][RTL8721CSM] Stability fixes and fixes for wlan amebad, updates to armv8m

### PR DESCRIPTION
**Pull request by prashantrar**
Fixing the hang issue seen during iperf test.
- Wlan is stable now.
- IPERF test works both in client and server mode
- OS is stable.
- Tasks are able to run smoothly.
- Wifi manager is completed and running.

Testing:
- Tested wifi connectivity and IPERF
- Tested GPIO functionality.

Things to do:
- Improve wlan performance

### Update

There are 2 sets of major changes that will be included in this PR

1. Changes to update the armv8m to the latest and update configs for armv8m MPU.(0cc8feb)
2. changes to fix the stability issues in WLAN library.(3e6e8b8)

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [b075de6388c6e3370cf74945d9e3989b318fe2c4] - Code Rule Check (C++) OK.

**Comment by jeongchanKim** [comment]
I think you'd better separate the commits which only has related files.

File: framework/src/wifi_manager/wifi_manager.c [hunk 1]
```diff
@@ -662,6 +662,7 @@ wifi_manager_result_e _wifimgr_run_sta(void)
 wifi_manager_result_e _wifimgr_save_connected_config(wifi_manager_ap_config_s *config)
 {
 	WM_LOG_START;
+
```
Comment (sunghan-chang) [inline framework/src/wifi_manager/wifi_manager.c]: Could you revert this change?

File: os/arch/arm/src/amebad/amebad_irq.c [hunk 2]
```diff
@@ -335,27 +351,31 @@ void up_irqinitialize(void)
...
 
   /* If the MPU is enabled, then attach and enable the Memory Management
    * Fault handler.
    */
-
-#ifdef CONFIG_ARM_MPU
-  //irq_attach(AMEBAD_IRQ_MEMFAULT, up_memfault, NULL);
+#if 0
+#ifdef CONFIG_ARMV8M_MPU
+  irq_attach(AMEBAD_IRQ_MEMFAULT, up_memfault, NULL);
```
Comment (jeongchanKim) [inline os/arch/arm/src/amebad/amebad_irq.c]: Is this not needed? There is only `ifndef` case about MPU in the below.

File: os/arch/arm/src/amebad/amebad_timerisr.c [hunk 3]
```diff
@@ -144,9 +151,9 @@ void up_timer_initialize(void)
 
   putreg32((NVIC_SYSTICK_CTRL_CLKSOURCE | NVIC_SYSTICK_CTRL_TICKINT |
             NVIC_SYSTICK_CTRL_ENABLE), NVIC_SYSTICK_CTRL);
+  DiagPrintf("%08x %08x\r\n", getreg32(NVIC_SYSTICK_CTRL), getreg32(NVIC_SYSTICK_RELOAD));
```
Comment (jeongchanKim) [inline os/arch/arm/src/amebad/amebad_timerisr.c]: I think this should be changed to something dbg.

File: os/arch/arm/src/armv8-m/mpu-reg.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
```
Comment (jeongchanKim) [inline os/arch/arm/src/armv8-m/mpu-reg.h]: Please change the license year.

File: os/arch/arm/src/armv8-m/up_doirq.c [hunk 5]
```diff
@@ -101,22 +128,41 @@ uint32_t *up_doirq(int irq, uint32_t *regs)
...
+	 * switch occurred during interrupt processing.  Context switching should
+	 * only be performed when the outermost interrupt handler returns.
+	 */
+
+	flags = irqsave();
+
+	g_nestlevel--;
+
+	if (g_nestlevel == 0) {
+		regs = (uint32_t*)current_regs;
```
Comment (jeongchanKim) [inline os/arch/arm/src/armv8-m/up_doirq.c]: Please add space between `uint32_t` and `*`.

File: os/binfmt/binfmt_execmodule.c [hunk 6]
```diff
@@ -81,7 +81,7 @@
 
 #ifdef CONFIG_BINFMT_ENABLE
 
-#ifdef CONFIG_ARMV7M_MPU
+#if defined(CONFIG_ARMV7M_MPU) || defined(CONFIG_ARMV8M_MPU)
```
Comment (jeongchanKim) [inline os/binfmt/binfmt_execmodule.c]: How about making one config for representing MPU?
For example,
```
config ARM_MPU
   bool

config ARMV7M_MPU
  bool "afdaf"
  select ARM_MPU


config ARMV8M_MPU
  bool "afdaf"
  select ARM_MPU
``

File: os/board/rtl8721csm/src/component/os/os_dep/include/osdep_service.h [hunk 7]
```diff
@@ -1404,6 +1381,8 @@ struct osdep_service_ops {
 	int (*rtw_create_task)(struct task_struct *task, const char *name, u32 stack_size, u32 priority, thread_func_t func, void *thctx);
 	void (*rtw_delete_task)(struct task_struct *task);
 	void (*rtw_wakeup_task)(struct task_struct *task);
+	void (*rtw_suspend_task)(void* task);
```
Comment (jeongchanKim) [inline os/board/rtl8721csm/src/component/os/os_dep/include/osdep_service.h]: Please change like below.
`void *task`

File: os/board/rtl8721csm/src/component/os/os_dep/osdep_service.c [hunk 8]
```diff
@@ -1160,6 +1154,26 @@ void rtw_wakeup_task(struct task_struct *task)
 	return;	
 }
 
+void rtw_suspend_task(void * task)
+{
+	if(osdep_service.rtw_suspend_task)
```
Comment (jeongchanKim) [inline os/board/rtl8721csm/src/component/os/os_dep/osdep_service.c]: Please add space after `if`.

File: os/arch/arm/src/armv8-m/mpu.h [hunk 9]
```diff
@@ -90,9 +93,9 @@
 /* MPU Type Register Bit Definitions */
 
 #define MPU_TYPE_SEPARATE       (1 << 0)	/* Bit 0: 0:unified or 1:separate memory maps */
-#define MPU_TYPE_DREGION_SHIFT  (8)	/* Bits 8-15: Number MPU data regions */
+#define MPU_TYPE_DREGION_SHIFT  (8)			/* Bits 8-15: Number MPU data regions */
```
Comment (deepaksrma) [inline os/arch/arm/src/armv8-m/mpu.h]: maintain same tab space for comments

File: os/arch/arm/src/armv8-m/mpu.h [hunk 10]
```diff
@@ -90,9 +93,9 @@
...
 #define MPU_TYPE_DREGION_MASK   (0xff << MPU_TYPE_DREGION_SHIFT)
-#define MPU_TYPE_IREGION_SHIFT  (16)	/* Bits 16-23: Number MPU instruction regions */
+#define MPU_TYPE_IREGION_SHIFT  (16)		/* Bits 16-23: Number MPU instruction regions */
```
Comment (deepaksrma) [inline os/arch/arm/src/armv8-m/mpu.h]: same as above

File: os/binfmt/binfmt_execmodule.c (hunk 6 above)
Comment (prashantrar) [reply os/binfmt/binfmt_execmodule.c]: @jeongchanKim for this I plan to add the following lines to "os/arch/arm/Kconfig"

```
config ARMV8M_MPU
	bool "MPU support"
	default n
	depends on ARCH_HAVE_MPU
	select ARCH_USE_MPU
	---help---
		Build in support for the ARM Cortex-M33 Memory Protection Unit (MPU).
		Check your chip specifications first; not all Cortex-M33 chips support the MPU.

config ARMV8M_MPU_NREGIONS
	int "Number of MPU regions"
	default 8 if ARCH_CORTEXM33
	depends on ARMV8M_MPU
	---help---
		This is the number of protection regions supported by the MPU.
```

Please let me know if this is appropriate?

Comment (jeongchanKim) [reply os/binfmt/binfmt_execmodule.c]: I mean, we'd better have one representative config for enabling MPU.

Comment (prashantrar) [reply os/binfmt/binfmt_execmodule.c]: so you suggest to remove the filter and add one common filter like "#ifdef CONFIG_ARM_MPU"?

Comment (jeongchanKim) [reply os/binfmt/binfmt_execmodule.c]: Yes, right. Because, if another arch-MPU is added, it looks not good to add continously with '||'.

Comment (prashantrar) [reply os/binfmt/binfmt_execmodule.c]: Perfect, got it, ill make the changes in the Kconfig and the c file.

**Comment by seinfra** [comment]
Target : [fbf3f8854bae10ccc5c789eaf5beda563bdb6b91] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fbf3f8854bae10ccc5c789eaf5beda563bdb6b91] - Code Rule Check OK.

File: os/arch/arm/src/armv8-m/mpu.h (hunk 10 above)
Comment (prashantrar) [reply os/arch/arm/src/armv8-m/mpu.h]: Done, fixed in 19305da

File: os/arch/arm/src/armv8-m/mpu.h (hunk 9 above)
Comment (prashantrar) [reply os/arch/arm/src/armv8-m/mpu.h]: fixed in 19305da

File: os/binfmt/binfmt_execmodule.c (hunk 6 above)
Comment (prashantrar) [reply os/binfmt/binfmt_execmodule.c]: fixed in 19305da

File: os/arch/arm/src/armv8-m/mpu-reg.h (hunk 4 above)
Comment (prashantrar) [reply os/arch/arm/src/armv8-m/mpu-reg.h]: fixed in 19305da

File: os/arch/arm/src/armv8-m/up_doirq.c (hunk 5 above)
Comment (prashantrar) [reply os/arch/arm/src/armv8-m/up_doirq.c]: fixed in 19305da

File: os/arch/arm/src/amebad/amebad_irq.c (hunk 2 above)
Comment (prashantrar) [reply os/arch/arm/src/amebad/amebad_irq.c]: fixed in 19305da

**Comment by seinfra** [comment]
Target : [19305da5f184ef7e21decde8001153868f4a51bb] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [19305da5f184ef7e21decde8001153868f4a51bb] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@prashantrar You should not do like this.
```bash
$ git reset --hard HEAD~3                   (After this command, check that the "Stability fixes and fixes for wlan" is at top of history)
$ git reset --soft HEAD^
$ git reset HEAD ./                                  (change all of green to red (unstage))
$ git add os/arch/arm/src/amebad os/board/....     (add Realtek specific codes first)
$ git commit   (make 1st commit)
$ git add ....     (add remain for common)
$ git commit   (make 2nd commit)
$ git push -f    (forcely update)
```

**Comment by seinfra** [comment]
Target : [0cc8feb56ae58430f39d8f93e05920eeaf789849] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0cc8feb56ae58430f39d8f93e05920eeaf789849] - Code Rule Check OK.

File: os/board/rtl8721csm/src/component/os/os_dep/osdep_service.c (hunk 8 above)
Comment (prashantrar) [reply os/board/rtl8721csm/src/component/os/os_dep/osdep_service.c]: fixed in 3e6e8b8

File: os/board/rtl8721csm/src/component/os/os_dep/include/osdep_service.h (hunk 7 above)
Comment (prashantrar) [reply os/board/rtl8721csm/src/component/os/os_dep/include/osdep_service.h]: fixed in 3e6e8b8

File: os/arch/arm/src/amebad/amebad_timerisr.c (hunk 3 above)
Comment (prashantrar) [reply os/arch/arm/src/amebad/amebad_timerisr.c]: fixed in 3e6e8b8

File: os/arch/arm/src/amebad/amebad_irq.c (hunk 2 above)
Comment (prashantrar) [reply os/arch/arm/src/amebad/amebad_irq.c]: fixed in 3e6e8b8

Comment (prashantrar) [reply os/arch/arm/src/amebad/amebad_irq.c]: fixed in 3e6e8b8

File: framework/src/wifi_manager/wifi_manager.c (hunk 1 above)
Comment (prashantrar) [reply framework/src/wifi_manager/wifi_manager.c]: fixed in 3e6e8b8

**Comment by prashantrar** [comment]
> 
> 
> @prashantrar You should not do like this.
> 
> ```shell
> $ git reset --hard HEAD~3                   (After this command, check that the "Stability fixes and fixes for wlan" is at top of history)
> $ git reset --soft HEAD^
> $ git reset HEAD ./                                  (change all of green to red (unstage))
> $ git add os/arch/arm/src/amebad os/board/....     (add Realtek specific codes first)
> $ git commit   (make 1st commit)
> $ git add ....     (add remain for common)
> $ git commit   (make 2nd commit)
> $ git push -f    (forcely update)
> ```

@sunghan-chang , as requested by you, I have forced the 2 commits. Now there are only 2 commits with the required changes as discussed.

File: os/arch/arm/include/amebad/chip.h [hunk 11]
```diff
@@ -64,16 +64,21 @@
...
+ */
+#define NVIC_SYSH_PRIORITY_MIN     0xe0	/* Bits [5:7] set in minimum priority */
 #define NVIC_SYSH_PRIORITY_DEFAULT 0x80	/* Midpoint is the default */
 #define NVIC_SYSH_PRIORITY_MAX     0x00	/* Zero is maximum priority */
-#define NVIC_SYSH_PRIORITY_STEP    0x10	/* Four bits of interrupt priority used */
+#define NVIC_SYSH_PRIORITY_STEP    0x20	/* Three bits of interrupt priority used */
+
 
-#define ARMV8M_PERIPHERAL_INTERRUPTS 58
+#define ARMV8M_PERIPHERAL_INTERRUPTS 64//MAX_PERIPHERAL_IRQ_NUM
```
Comment (sangam-swami) [inline os/arch/arm/include/amebad/chip.h]: give tab for comments

File: os/arch/arm/src/armv8-m/Toolchain.defs [hunk 12]
```diff
@@ -224,9 +225,9 @@ ifeq ($(CONFIG_ARMV8M_TOOLCHAIN),CODEREDW)
   MAXOPTIMIZATION ?= -Os
   ifeq ($(CONFIG_ARCH_CORTEXM33),y)
     ifeq ($(CONFIG_ARCH_FPU),y)
-      ARCHCPUFLAGS = -mcpu=cortex-m33 -mthumb -mfpu=fpv4-sp-d16 -mfloat-abi=hard
+      ARCHCPUFLAGS = -mcpu=cortex-m33 -mthumb -mcmse -mfpu=fpv4-sp-d16 -mfloat-abi=hard
```
Comment (sangam-swami) [inline os/arch/arm/src/armv8-m/Toolchain.defs]: do we need to change to fpv5 for windows also?

File: build/configs/rtl8721csm/hello/defconfig [hunk 13]
```diff
@@ -846,7 +843,7 @@ CONFIG_ARCH_HAVE_HEAPCHECK=y
 # CONFIG_DEBUG_HEAP is not set
 CONFIG_DEBUG_MM_HEAPINFO=y
 # CONFIG_DEBUG_CHECK_FRAGMENTATION is not set
-# CONFIG_DEBUG_IRQ is not set
+# CONFIG_DEBUG_IRQ
```
Comment (jsdosa) [inline build/configs/rtl8721csm/hello/defconfig]: Is this correct?

File: os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.h [hunk 14]
```diff
@@ -77,59 +77,66 @@ extern VOID RtlUdelayOS(u32 us);
...
 #warning "DLIST_NOT_DEFINE!!!!!!"
 struct list_head {
 	struct list_head *next, *prev;
 };
 #endif
 
-struct __queue {
-	struct list_head queue;
-	_lock lock;
+struct	__queue	{
```
Comment (jsdosa) [inline os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.h]: Is this the correct tab?

File: os/arch/arm/src/armv8-m/Toolchain.defs (hunk 12 above)
Comment (prashantrar) [reply os/arch/arm/src/armv8-m/Toolchain.defs]: we are just adding the cmse flag here

File: os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.h (hunk 14 above)
Comment (prashantrar) [reply os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.h]: Hey I have double checked this, it seems to be the right tab. Should a tab be 1 space or 4 spaces?

File: build/configs/rtl8721csm/hello/defconfig (hunk 13 above)
Comment (prashantrar) [reply build/configs/rtl8721csm/hello/defconfig]: fixed in latest commit

File: os/arch/arm/include/amebad/chip.h (hunk 11 above)
Comment (prashantrar) [reply os/arch/arm/include/amebad/chip.h]: fixed in latest commit

**Comment by seinfra** [comment]
Target : [1ff0da5acf78cafc59ebb2326d538622c7d42d1d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1ff0da5acf78cafc59ebb2326d538622c7d42d1d] - Code Rule Check OK.

File: os/arch/arm/src/armv8-m/Toolchain.defs (hunk 12 above)
Comment (sangam-swami) [reply os/arch/arm/src/armv8-m/Toolchain.defs]: this is related to floating-point h/w available on target. how it varies for different build platforms?

**Comment by seinfra** [comment]
Target : [43d0875aeaf16ecca14fd065329c2628ca039378] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [43d0875aeaf16ecca14fd065329c2628ca039378] - Code Rule Check OK.

File: os/arch/arm/src/armv8-m/Toolchain.defs (hunk 12 above)
Comment (prashantrar) [reply os/arch/arm/src/armv8-m/Toolchain.defs]: This is fixed in latest commit. You are right the fpu needs to be common, I got confused betweeen the fpu flag and cmse

File: os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.h (hunk 14 above)
Comment (prashantrar) [reply os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.h]: @jsdosa any update on this?

Comment (prashantrar) [reply os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.h]: @jsdosa any update on this?

**Comment by seinfra** [comment]
Target : [fbc642fa3bb965a1c0bec68f80915a823cc299f4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fbc642fa3bb965a1c0bec68f80915a823cc299f4] - Code Rule Check Result:
os/se/ameba/security_ameba_wrapper.c:205: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:205: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:205: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:270: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:270: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:270: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:359: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:359: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:359: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:472: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:472: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:472: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:475: ERROR: [SPC_M_KWD] missing space after struct definition
os/se/ameba/security_ameba_wrapper.c:713: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/se/ameba/security_ameba_wrapper.c:808: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:816: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:824: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:832: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:1200: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1201: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1202: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1276: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1277: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1278: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1325: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1326: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1340: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'

**Comment by sunghan-chang** [comment]
@prashantrar Could you check below?

1. execute below
```shell
$ git rebase -i HEAD~5
pick Amebad board related changes for wlan stability
pick Adding changes to common files and bring armv8m upto date
pick fixing review comments
pick fixing incorrect compiler flag
pick security API for Ameba rtl8721csm
```
2. move 3rd and 4th to 2nd line
```shell
    pick Amebad board related changes for wlan stability
    pick fixing review comments
    pick fixing incorrect compiler flag
    pick Adding changes to common files and bring armv8m upto date
    pick security API for Ameba rtl8721csm
```
3. change the word for 2nd and 3rd as fixup
```shell
    pick Amebad board related changes for wlan stability
    fixup fixing review comments
    fixup fixing incorrect compiler flag
    pick Adding changes to common files and bring armv8m upto date
    pick security API for Ameba rtl8721csm
```
4. exit by ^x key

5. push the changes
```shell
$ git push -f
```

**Comment by sunghan-chang** [comment]
@prashantrar Then they will have 3 commits, 
```shell
    Amebad board related changes for wlan stability
    Adding changes to common files and bring armv8m upto date
    security API for Ameba rtl8721csm
```

**Comment by seinfra** [comment]
Target : [1932a56d8fad3d8c9191e758dcd0e4d13a4ee39b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1932a56d8fad3d8c9191e758dcd0e4d13a4ee39b] - Code Rule Check Result:
os/se/ameba/security_ameba_wrapper.c:205: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:205: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:205: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:270: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:270: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:270: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:359: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:359: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:359: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:472: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:472: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:472: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:475: ERROR: [SPC_M_KWD] missing space after struct definition
os/se/ameba/security_ameba_wrapper.c:713: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/se/ameba/security_ameba_wrapper.c:808: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:816: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:824: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:832: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/se/ameba/security_ameba_wrapper.c:1200: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1201: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1202: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1276: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1277: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1278: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1325: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1326: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/se/ameba/security_ameba_wrapper.c:1340: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'

**Comment by prashantrar** [comment]
@sunghan-chang as requested i have squased the commits and rebased as you mentioned.

**Comment by sunghan-chang** [comment]
@prashantrar I am sorry but could you make an another PR for SE change?
```shell
$ git branch master
$ git checkout -b new_branch_for_se
$ git cherry-pick 1932a56
$ git push -u origin new_branch_for_se
make a new PR on github.com for SE
$ git checkout wlan_stability_fix
$ git reset --hard HEAD^
$ git push -f
```

**Comment by prashantrar** [comment]
@sunghan-chang as mentioned, ive cherry picked the commit and made a new PR #4236

**Comment by sunghan-chang** [comment]
@prashantrar I am waiting to merge but required checks are not working. Could you do force-push again to refresh them?

**Comment by prashantrar** [comment]
@sunghan-chang  I did a force push, it was not working, I even tried to merge with master and put another commit, the checks are still not happening, They are also not happening for the new PR that I raised

**Comment by seinfra** [comment]
Target : [92c591bf13bd2b9e3b3e8c9eae25ddf4e9260586] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [92c591bf13bd2b9e3b3e8c9eae25ddf4e9260586] - Code Rule Check Result:
framework/include/media/BufferOutputDataSource.h:42: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/BufferOutputDataSource.h:72: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)

**Comment by seinfra** [comment]
Target : [1223b8332ef5f5fbf57ece02dcf684b529997aba] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1223b8332ef5f5fbf57ece02dcf684b529997aba] - Code Rule Check Result:
framework/include/media/BufferOutputDataSource.h:42: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/include/media/BufferOutputDataSource.h:72: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)

**Comment by sunghan-chang** [comment]
@prashantrar You should not add your private history into public. What I am saying is the commit, `Merge branch 'master' into wlan_stability_fix`. It is merging the latest codes into your private branch. When you need to update yours as latest, you should use the `rebase` command instead of `merge`. 
Could you remove the last commit?
```shell
$ git reset --hard HEAD^
$ git push -f
```

**Comment by sunghan-chang** [comment]
When a PR has just a kind of change, I can merge it including your private commit. Because github supports the `squash and merge` option, I can remove private commit. But this PR already has two different kinds of changes so that I can't use it. That's why I request removing it by you.

**Comment by prashantrar** [comment]
> 
> 
> @prashantrar You should not add your private history into public. What I am saying is the commit, `Merge branch 'master' into wlan_stability_fix`. It is merging the latest codes into your private branch. When you need to update yours as latest, you should use the `rebase` command instead of `merge`.
> Could you remove the last commit?
> 
> ```shell
> $ git reset --hard HEAD^
> $ git push -f
> ```

I actually did this yesterday to make travis run again but even then it didnt run. I will try again now

**Comment by prashantrar** [comment]
I just did it, if you notice travis is not running now

**Comment by seinfra** [comment]
Target : [14397d732a8ed6af69e99c778596eaf7b714d08f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [07cd652673a826746df914d7d2d3f5b859ac0f20] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [07cd652673a826746df914d7d2d3f5b859ac0f20] - Code Rule Check (C++) OK.

File: build/configs/rtl8721csm/tc/defconfig [hunk 15]
```diff
@@ -136,15 +137,24 @@ CONFIG_BOARD_LOOPSPERMSEC=16717
...
+# CONFIG_ARCH_HAVE_INTERRUPTSTACK is not set
 # CONFIG_ARCH_HAVE_NESTED_INTERRUPT is not set
-CONFIG_ARCH_HAVE_HIPRI_INTERRUPT=y
-# CONFIG_ARCH_HIPRI_INTERRUPT is not set
+# CONFIG_ARCH_HAVE_HIPRI_INTERRUPT is not set
+
+#
+# Boot Memory Configuration
+#
+CONFIG_RAM_REGIONx_START="0x0,0x0,"
```
Comment (jeongchanKim) [inline build/configs/rtl8721csm/tc/defconfig]: Please set the proper physical RAM address.

File: build/configs/rtl8721csm/tc/defconfig [hunk 16]
```diff
@@ -175,8 +192,9 @@ CONFIG_ARCH_USE_FLASH=y
 #
 CONFIG_FLASH_PARTITION=y
 CONFIG_FLASH_MINOR=0
-CONFIG_FLASH_PART_SIZE="4096,"
+CONFIG_FLASH_PART_SIZE="128,"
```
Comment (jeongchanKim) [inline build/configs/rtl8721csm/tc/defconfig]: Is it okay to set the smartfs part only?

File: build/configs/rtl8721csm/tc/defconfig (hunk 15 above)
Comment (prashantrar) [reply build/configs/rtl8721csm/tc/defconfig]: We cannot hard-code this address in defconfig, this is determined by the size of the application and after the linker has successfully run. The start address and size is initialized in os_heap_init in rtl8721dhp_app_start.c

File: build/configs/rtl8721csm/tc/defconfig (hunk 16 above)
Comment (prashantrar) [reply build/configs/rtl8721csm/tc/defconfig]: Because we use different flash memories in the development board, so the flash address for smartfs and also the partition size needs to be determined by the user.
